packages/context/src/loader.ts

```
@@ -0,0 +1,252 @@
+): Promise<CAPProject | undefined> {
+  if (projectInfo.kind === "Java") {
+    console.log("Java project is not supported yet.");
+    return;
+  }
+  const app = await getApp(
+    projectRoot,
+    appRoot,
+    manifest,
+    manifestDetails,
+    projectInfo
+  );
+  if (!app) {
+    return;
+  }
+  return {
+    type: CAP_PROJECT_TYPE,
+    kind: "NodeJS",
+    root: projectRoot,
+    apps: new Map([[appRoot, app]]),
+  };
+}
+/**
+ * Get UI5 project
+ *
+ * @param projectRoot path to root of a project
+ * @param appRoot path to root of an app
+ * @param manifest path to manifest defined in under app root
+ * @param manifestDetails details of manifest
+ */
+export async function getUI5Project(
+  projectRoot: string,
+  appRoot: string,
+  manifest: Manifest,
+  manifestDetails: ManifestDetails
+): Promise<UI5Project | undefined> {
+  const app = await getApp(projectRoot, appRoot, manifest, manifestDetails, {
+    kind: "UI5",
+    type: "UI5",
+  });
+  if (!app) {
+    return;
+  }
+  return {
+    type: UI5_PROJECT_TYPE,
+    root: projectRoot,
+    app,
+  };
+}
+
+/**
+ * Get CAP or UI5 project
+ *
+ * @param documentPath path to a file i.e absolute/path/webapp/ext/main/Main.view.xml
+ */
+export async function getProject(
+  documentPath: string
+): Promise<Project | undefined> {
+  const projectRoot = await getProjectRoot(documentPath);
+  if (!projectRoot) {
+    return;
+  }
+  const cachedProject = cache.getProject(projectRoot);
+  if (cachedProject) {
+    return cachedProject;
+  }
+  const appRoot = await findAppRoot(documentPath);
+  if (!appRoot) {
+    return;
+  }
+  const manifestPath = await findManifestPath(documentPath);
+  if (!manifestPath) {
+    return;
+  }
+  const manifest = await getUI5Manifest(manifestPath);
+  if (!manifest) {
+    return;
+  }
+  const manifestDetails = await getManifestDetails(documentPath);
+  if (!manifestDetails) {
+    return;
+  }
+  const projectInfo = await getProjectInfo(projectRoot);
```

I'd suggest reject promise with specific result (code+description) instead of returning undefined in all above cases. This would help the consumer to understand where the problem actually is.

---

That is a good idea and further improvement maybe as per need.